### PR TITLE
Add support for GHC 8.10.1 

### PR DESCRIPTION
**Pull request by bubba**
Continuing the work of @jneira in #305 , rebased

### CODE REVIEW

**Comment by codygman** [comment]
Related to #269

**Comment by tfausak** [comment]
This doesn't build with GHC 8.4, 8.2, or 8.0. Do you know if it's an easy fix or not? 

- 8.4: https://github.com/lspitzner/brittany/pull/324/checks?check_run_id=1402854550
- 8.2: https://github.com/lspitzner/brittany/pull/324/checks?check_run_id=1402854557
- 8.0: https://github.com/lspitzner/brittany/pull/324/checks?check_run_id=1402854564

**Comment by jneira** [comment]
@tfausak maybe i am not interpreting those builds correctly but they seem fail for ghc-8.10.2 and no for other versions??

**Comment by jneira** [comment]
Afaics all changes in this pr was made under cpp conditions so it should not interfere with older ghc versions (module some error)

**Comment by tfausak** [comment]
I think some of the CPP changes looked like this:

``` hs
# if MIN_VERSION_ghc(8, 10, 1)
-- nothing
# else
some existing code
# endif
```

File: src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs [hunk 1]
```diff
@@ -1137,12 +1154,15 @@ layoutExpr lexpr@(L _ expr) = do
     HsStatic{} -> do
       -- TODO
       briDocByExactInlineOnly "HsStatic{}" lexpr
+#if MIN_VERSION_ghc(8,10,1)   /* ghc-8.10.1 */
```
Comment (jneira) [inline src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs:1157]: cpp if branch without code

File: src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs [hunk 2]
```diff
@@ -1152,6 +1172,8 @@ layoutExpr lexpr@(L _ expr) = do
     HsTickPragma{} -> do
       -- TODO
       briDocByExactInlineOnly "HsTickPragma{}" lexpr
+#if MIN_VERSION_ghc(8,10,1)   /* ghc-8.10.1 */
```
Comment (jneira) [inline src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs:1175]: cpp if branch without code

**Comment by jneira** [comment]
Yeah, but it should no make fail prevoius ghc versions but make ghc-8.10.1 fail, no?
(sorry if ai am woring i am only gathering at the code without even compiling it)

Can you share the errors that were thrown for earlier versions? are there in the travis failed build for this pr?

**Comment by tfausak** [comment]
Sorry, my comment wasn't very helpful. Also apparently the links I shared earlier don't work anymore. I probably shouldn't have rebased #324. Here are actually usable links to the build failures: 

- 8.4: https://github.com/lspitzner/brittany/runs/1402854550
- 8.2: https://github.com/lspitzner/brittany/runs/1402854557
- 8.0: https://github.com/lspitzner/brittany/runs/1402854564

**Comment by tfausak** [comment]
I'm closing this in favor of #324, which includes these commits. Please let me know if I've missed something!

**Pull request closed without merging**